Core/Graphics/Source/Graphics.cpp

```diff
@@ -88,8 +94,15 @@ namespace Babylon
         }
         else
         {
-            return whenAllTask.then(Dispatcher, arcana::cancellation::none(), [this, &finished, &workDone]() mutable {
-                return RenderCurrentFrameAsync(finished, workDone);
+            return whenAllTask.then(Dispatcher, arcana::cancellation::none(), [this, &finished, &workDone, &error](const arcana::expected<void, std::exception_ptr>& result) mutable {
+                if (result.has_error())
+                {
+                    finished = true;
+                    error = result.error();
```

Comment:
Why is this needed? Can't the caller just get it in a task continuation?

Reply:
> Can't the caller just get it in a task continuation?

No. I thought the same thing, but this code doesn't return the last task of the chain. It returns the first task of the chain.